find_email_address.py

```diff
@@ -0,0 +1,123 @@
+from selenium import webdriver
+from pprint import pprint
+from urlparse import urlparse
+import argparse
+
+parser = argparse.ArgumentParser(description='Scrape a domain for email addresses.')
+parser.add_argument('hostname', metavar='domain', nargs=1,
+                    help='Host to scrape')
+parser.add_argument('--port',
+                    help='Port number, default: 80')
+parser.add_argument('--chrome',
+                    help='Use Chrome instead of HTMLUNIT',  action="store_true")
+parser.add_argument('--verbose',
+                    help='Output additional logging',  action="store_true")
+#parser.add_argument('--depth',
+#                    help='Unimplemented: Max number of pages to fetch, default: 100')
+
+args = parser.parse_args()
+
+domain = args.hostname[0]
+
+if args.port is None:
+    port = 80
+else:
+    port = args.port
+
+if args.chrome:
+    driver = webdriver.Chrome()
+else:
+    driver = webdriver.Remote(desired_capabilities=webdriver.DesiredCapabilities.HTMLUNIT)
+
+def log(msg):
+    if args.verbose:
+        print(msg)
+
+def find_links(driver, domain):
+    retval = {'emails': set(), 'links': set()} # emails for mailto: links, links for everything else
+    elems = driver.find_elements_by_xpath('//a[@href]')
+    # could use //a[startswith(href'mailto')] to get emails and do email links
+    # separately but we'll just get all links and then divvy them up
+    for el in elems:
+        link = el.get_attribute('href')
+        if (link is None):
+            continue
+        url = urlparse(link)
+
+        if url.scheme.startswith('mailto'):
+            log('Email found: ' + url.path)
+            retval['emails'].add(url.path)
+        else:
+            #pprint(url)
+
+            if url.netloc.startswith(domain):
```

Comment:
Whoops, I hadn't read far enough in the documentation for urlparser actually.  New PR uses url.hostname.  netloc included port number so used startwith.